src/index.ts

```diff
@@ -0,0 +1,31 @@
+import * as grpc from 'grpc';
+import { Metadata } from 'grpc';
+import { promisify } from 'util';
+
+export { status, ClientReadableStream } from 'grpc';
+export * from 'google-protobuf/google/protobuf/timestamp_pb.js';
+export * from '../proto/compiled/logproto_grpc_pb';
+export * from '../proto/compiled/logproto_pb';
+
+export interface IGrpcClientAsync {
+	[key: string]: IGrpcMethod;
+}
+
+export type IGrpcMethod = (data: any, metadata?: Metadata) => Promise<any>;
+
+export function promisifyClient(client: any): IGrpcClientAsync {
+	return Object.values<string>(client.$method_names).reduce((acc, name) => {
+		acc[name] = promisify(client[name].bind(client));
+		return acc;
+	}, {} as IGrpcClientAsync);
```

Comment:
I find the iterative style clearer for reduce cases like this where the same `acc` is always returned:
```
const promisifiedClient = {} as IGrpcClientAsync;
for (const name of Object.values<string>(client.$method_names)) {
	promisifiedClient[name] = promisify(client[name].bind(client));
}
return promisifiedClient

Reply:
Good point, updated.